hkmc2/shared/src/main/scala/hkmc2/semantics/Resolver.scala

```diff
@@ -559,18 +569,22 @@ class Resolver(tl: TraceLogger)
         case S(defn @ CallableDefinition(params = ps :: pss)) =>
           val (argCountUB, argCountLB) = as match
           // Tup: regular arguments
-          case tup: Term.Tup => (
-            !tup.fields.exists(_.isInstanceOf[Spd]),
-            tup.fields.map:
-              case Fld(asc = S(_)) => 0
-              case _: Fld => 1
-              case _: Spd => 0
-            .sum +
-            tup.fields.exists:
-              case Fld(asc = S(_)) => true
-              case _ => false
-            .into(if _ then 1 else 0)
-          )
+          case tup: (Term.Tup | Term.CtxTup) => 
+            val fields = tup match
+              case Term.Tup(fs) => fs
+              case Term.CtxTup(fs) => fs
+            (
+              !fields.exists(_.isInstanceOf[Spd]),
+              fields.map:
+                case Fld(asc = S(_)) => 0
+                case _: Fld => 1
+                case _: Spd => 0
```

Comment:
Is this case really sound? Maybe fix it now? Or add it to the issue tracker (if not already there).

Reply:
I don't see any unsoundness here? It's just the lower bound of argument

Reply:
Ah, I see. In that case, it's probably fine.